overlord/snapstate/backend/link.go

```diff
@@ -153,10 +149,28 @@ func updateCurrentSymlinks(info *snap.Info) (revert func(), e error) {
 	return revertFunc, nil
 }
 
+// MaybePrepareReboot configures the system for a reboot if necesssary because
+// of a snap refresh. isUndo must be set when we are installing the previous
+// snap while performing a revert of the latest one that was installed
+func (b Backend) MaybePrepareReboot(info *snap.Info, dev snap.Device, isUndo bool) (boot.RebootInfo, error) {
```

Comment:
Would using a `boot.NextBootContext` as the param rather than the `isUndo` boolean be an option? I guess it doesn't matter too much.

Reply:
Could be, but sounds a bit like exposing details that the caller does not need to know